src/uct/ib/rc/accel/rc_mlx5_common.h

```diff
@@ -90,6 +90,36 @@ UCT_RC_MLX5_DECLARE_ATOMIC_LE_HANDLER(32)
 UCT_RC_MLX5_DECLARE_ATOMIC_LE_HANDLER(64)
 
 
+#define UCT_RC_MLX5_IFACE_COMMON_INIT(_self, _md, _worker, _params,             \
+                                      _rc_config, _rc_common_config,            \
+                                      _tl_ops, _iface_ops, _fc_req_size, _flags)\
+({                                                                              \
+    uct_rc_mlx5_iface_init_attr_t _init_attr = {};                              \
+                                                                                \
+    _init_attr.super.fc_req_size           = _fc_req_size;                      \
+    _init_attr.super.flags                 = UCT_IB_CQ_IGNORE_OVERRUN;          \
+    _init_attr.super.cq_len[UCT_IB_DIR_TX] = (_rc_config)->tx_cq_len;           \
+    _init_attr.super.qp_type               = IBV_QPT_RC;                        \
+    _init_attr.super.max_rd_atomic         = IBV_DEV_ATTR(&(_md)->super.dev,    \
+                                                          max_qp_rd_atom);      \
+    _init_attr.flags                       = _flags;                            \
+                                                                                \
+    uct_ib_mlx5_parse_cqe_zipping(_md, &(_rc_common_config)->super,             \
+                                  &(_init_attr).super);                         \
+                                                                                \
+    UCS_CLASS_CALL_SUPER_INIT(uct_rc_mlx5_iface_common_t, _tl_ops, _iface_ops,  \
```

Comment:
can we move the common code inside uct_rc_mlx5_iface_common_t constructor, instead of introducing such macro?

Reply:
it will affect `dc` transport since `uct_dc_mlx5_iface_t` ctor calls `uct_rc_mlx5_iface_common_t` ctor too.

Reply:
because `UCS_CLASS_` defines its own infrastructure variables which becomes out of scope of the function but required to be passed to base ctor